mapper-runtime/src/main/java/com/datastax/oss/driver/api/mapper/entity/naming/NamingConvention.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.api.mapper.entity.naming;
+
+import com.datastax.oss.driver.api.mapper.annotations.Entity;
+import com.datastax.oss.driver.api.mapper.annotations.NamingStrategy;
+
+/**
+ * A built-in convention to infer CQL column names from the names used in an {@link
+ * Entity}-annotated class.
+ *
+ * @see NamingStrategy
+ */
+public enum NamingConvention {
```

Comment:
Unlike mapper 3 there is no notion of a "Java convention", it expects camel-case, period. I don't think that complexity was worth it, it's very unlikely that someone would name their methods `getProduct_Id`.

Also I'm open to suggestions for other conventions, but I think we should be conservative (even some of those are already a bit exotic IMO).